x-pack/plugins/fleet/server/routes/agent/handlers.ts

```diff
@@ -354,9 +354,17 @@ function isStringArray(arr: unknown | string[]): arr is string[] {
 
 export const getAvailableVersionsHandler: RequestHandler = async (context, request, response) => {
   try {
-    const availableVersions = await AgentService.getAvailableVersions({});
+    const availableVersions = await AgentService.getAvailableVersions();
```

Comment:
should we implement an in-memory cache instead? it's used in a few other place in the code to show upgradable agents and I am wondering what will be the impact here

Reply:
Yeah my initial implementation was an in-memory cache but it seemed like HTTP caching might be easier and accomplish the same thing. I guess using an in-memory cache does give us the escape hatch of restarting Kibana to clear the cache if we need to debug.

Reply:
could we use an in memory cache with expiration like `lru-cache` ? as HTTP cache will only work for that endpoint not for the other use case where we retrieve versions

Reply:
lru-cache doesn't seem like a great fit because it's more for evicting items as the cache size grows and less for TTL style behavior. I will implement a basic in memory caching solution here.

Reply:
In memory cache with 1 hour expiration has been added. Testing various upgrade behaviors around the app now with this change.